private.tf

```diff
@@ -8,12 +8,23 @@ module "private_label" {
   tags       = "${var.tags}"
 }
 
+module "private_subnet_label" {
+  source     = "git::https://github.com/cloudposse/tf_label.git?ref=tags/0.2.0"
+  namespace  = "${var.namespace}"
+  stage      = "${var.stage}"
+  name       = "private"
+}
+
 resource "aws_subnet" "private" {
   count             = "${length(var.availability_zones)}"
   vpc_id            = "${data.aws_vpc.default.id}"
   availability_zone = "${element(var.availability_zones, count.index)}"
   cidr_block        = "${cidrsubnet(signum(length(var.cidr_block)) == 1 ? var.cidr_block : data.aws_vpc.default.cidr_block, ceil(log(length(data.aws_availability_zones.available.names) * 2, 2)), count.index)}"
-  tags              = "${module.private_label.tags}"
+  tags              = {
+    "Name"          = "${module.private_subnet_label.id}-${element(var.availability_zones, count.index)}"
```

Comment:
Let's replace `-` with `var.delimiter` and `-` sign occurrences in `element(var.availability_zones, count.index)` to `var.delimiter`.